app/components/Exchange/TradingViewPriceChart.jsx

```diff
@@ -147,18 +176,95 @@ export default class TradingViewPriceChart extends React.Component {
         this.props.dataFeed.clearSubs();
     }
 
-    shouldComponentUpdate(np) {
-        if (np.chartHeight !== this.props.chartHeight) return true;
-        if (!!this.tvWidget) return false;
-        if (!np.marketReady) return false;
-        return true;
+    shouldComponentUpdate(np, state) {
+        return (
+            state.showLoadModal !== this.state.showLoadModal ||
+            state.showSaveModal !== this.state.showSaveModal ||
+            np.chartHeight !== this.props.chartHeight ||
+            this.props.charts.size !== np.charts.size ||
+            this.tvWidget ||
```

Comment:
Please double check the logic when refactoring and adjust the mistake here

Reply:
sorry, fixed